src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+ 
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+ 
+/** Stores blob keys for plant images in Datastore */
```

Comment:
This is an upload handler, right? As described here: https://cloud.google.com/appengine/docs/standard/java/blobstore#3_implement_upload_handler

Please document that this servlet is a blobstore upload handler. (I only figured it out because you used BlobstoreService.getUploads().)

Reply:
I added the words upload handler, but I wanted to make sure there's nothing else it's doing that I'm missing. Is that first chunk of code just getting the most recently uploaded blobkey or is it manipulating the upload?

Reply:
Why are you asking me? You wrote the code. You are responsible for knowing what it does and communicating the important parts to other engineers.

Reply:
Sorry, that came out a little harsh. I don't mean that you have to understand everything without talking to other engineers. I just mean that the time to ask questions and figure this stuff out is before sending out a PR.

Reply:
Yeah, I kind of figured that while I was typing it. I thought I knew what it was doing (I assumed it was just grabbing the blobkey after it was placed by the uploadBlobstore servlet), but parts of the documentation you sent went over my head and I realized that might not be the case. In the future, I'll try to catch that before I send the pull request. Since this is already out, does my understanding make sense (does the new description make sense?)

Reply:
Your description is accurate and concise, and maybe a little terse. It might help to link to the documentation for blobstore upload handlers, because Liyah/Hannah/Devon might not know what that means otherwise.

Reply:
Sounds good, I'll do that.